src/storage/src/hummock/compactor/shared_buffer_compact.rs

```diff
@@ -427,13 +428,16 @@ pub async fn merge_imms_in_memory(
     }
 
     let mut merged_payload: Vec<SharedBufferVersionedEntry> = Vec::new();
-    let mut pivot = items
+    let first_item_key = items
```

Comment:
maybe off topic: can we assert that items is non-empty instead of creating an implicit default value?

Reply:
I planned to do that before but I later realized it is possible that the items are empty when imm only contains range tombstone.

However, I am planning to disallow imm merge when there are range tombstones in imm, which can simplify the logic a lot.